extensions/amp-onetap-google/0.1/amp-onetap-google.js

```diff
@@ -0,0 +1,322 @@
+  border: 'none',
+  margin: 0,
+  padding: 0,
+  position: 'fixed',
+  right: 'auto',
+  left: '0',
+  top: 'auto',
+  bottom: '0',
+};
+
+/** @const {Object} */
+const elementStyle = {
+  height: '0',
+  width: '0',
+  'z-index': 9999,
+  border: 'none',
+  position: 'fixed',
+  left: '0',
+  bottom: '0',
+};
+
+export class AmpOnetapGoogle extends AMP.BaseElement {
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+
+    this.checkExperimentEnabled_();
+
+    /** @private @const {string} */
+    this.intermediateIframeUrl_ = this.element.getAttribute('data-iframe_url');
+
+    /** @private @const {string} */
+    this.intermediateIframeOrigin_ = new URL(
+      this.intermediateIframeUrl_
+    ).origin;
+
+    /** @private {!Element} */
+    this.doc_ = this.element.ownerDocument;
+
+    /** @private {boolean=} */
+    this.isBottomSheetUiMode_ = undefined;
+
+    /** @private {!Element=} */
+    this.intermediateIframe_;
+
+    /** @private {boolean} */
+    this.isIntermediateIframeVisible_ = false;
+
+    /** @private {boolean} */
+    this.cancelOnTapOutside_ = false;
+
+    /** @private */
+    this.onClickOutside_ = () => {
+      if (this.cancelOnTapOutside_ && this.isIntermediateIframeVisible_) {
+        this.doc_.removeEventListener('click', this.onClickOutside_);
+        this.removeIntermediateIframe_();
+      }
+    };
+  }
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return layout == Layout.CONTAINER;
+  }
+
+  /** @override */
+  buildCallback() {
+    console.log('buildCallback');
+    this.win.addEventListener('message', (event) => {
+      // Make sure the postMessage comes from the iframe origin
+      if (
+        this.validateMessageOrigin_(event.origin) &&
+        this.validateMessageSource_(event.source)
+      ) {
+        this.handleIntermediateIframeMessage_(event.data);
+      }
+    });
+    this.win.addEventListener('load', () => {
+      this.loadIntermediateIframe_();
+    });
+  }
+
+  /** @override */
+  layoutCallback() {
+    console.log('layoutCallback');
+  }
+
+  /** @override */
+  onLayoutMeasure() {
+    console.log('onLayoutMeasure');
+  }
+
+  /** @override */
+  unLayoutCallback() {
+    console.log('unLayoutCallback');
+  }
+
+  /**
+   * @private
+   * @param {string} origin
+   * @return {boolean} if the origin is valid.
+   */
+  validateMessageOrigin_(origin) {
+    return origin === this.intermediateIframeOrigin_;
+  }
+
+  /**
+   * @private
+   * @param {Object} source
+   * @return {boolean} if the message source is valid.
+   */
+  validateMessageSource_(source) {
+    return (
+      this.intermediateIframe_ &&
+      this.intermediateIframe_.contentWindow &&
+      this.intermediateIframe_.contentWindow == source
+    );
+  }
+
+  /**
+   * @private
+   * @param {Object} data
+   */
+  handleIntermediateIframeMessage_(data) {
+    if (!this.intermediateIframe_) {
+      return;
+    }
+    switch (data['command']) {
```

Comment:
any time you mutate the dom use a mutate promise from vsync so AMP knows you've changed somthing and can relayout other components if needed - see https://github.com/ampproject/amphtml/blob/master/extensions/amp-subscriptions/0.1/dialog.js for an example

Reply:
Changed.